src/app/frontend/common/services/global/kubeconfig.ts

```diff
@@ -0,0 +1,246 @@
+    private readonly state_: Router,
+    @Inject(CONFIG_DI_TOKEN) private readonly CONFIG: Config,
+  ) {}
+
+  setKubeconfig(configYaml: string): void {
+    if (!configYaml.length) {
+      return null;
+    }
+    // Store kubeconfig from session storage
+    this.sessionStorage_.store(this.CONFIG.kubeconfigSessionStorageName, configYaml);
+    // Store API server endpoint into cookie
+    this.cookies_.set(this.CONFIG.serverCookieName, this.getCurrentServer(), null, null, null, false, 'Strict');
+    // Store metrics-scraper endpoint endpoint into cookie
+    this.cookies_.set(
+      this.CONFIG.sidecarHostCookieName,
+      this.getCurrentSidecarHost(),
+      null,
+      null,
+      null,
+      false,
+      'Strict',
+    );
+  }
+
+  clearKubeconfig(): void {
+    // Clear kubeconfig from session storage
+    this.sessionStorage_.clear(this.CONFIG.kubeconfigSessionStorageName);
+    this.cookies_.delete(this.CONFIG.serverCookieName, null, null, false, 'Strict');
+    this.cookies_.delete(this.CONFIG.sidecarHostCookieName, null, null, false, 'Strict');
+  }
+
+  getKubeconfig(): string {
+    // Load kubeconfig from session storage
+    return this.sessionStorage_.retrieve(this.CONFIG.kubeconfigSessionStorageName) as string;
+  }
+
+  getContexts(): string[] {
+    const kubeconfig = this.sessionStorage_.retrieve(this.CONFIG.kubeconfigSessionStorageName) as string;
+    if (kubeconfig === null) {
+      return [];
+    }
+    const conf: Kubeconfig = loadYaml(kubeconfig);
+    return conf.contexts.map(ctx => {
+      return ctx.name;
+    });
+  }
+
+  getCurrentContext(): string {
+    const kubeconfig = this.sessionStorage_.retrieve(this.CONFIG.kubeconfigSessionStorageName) as string;
+    if (kubeconfig === null) {
+      return '';
+    }
+    const conf: Kubeconfig = loadYaml(kubeconfig);
+    return conf['current-context'];
+  }
+
+  getContext(contextName: string): Context {
+    const kubeconfig = this.sessionStorage_.retrieve(this.CONFIG.kubeconfigSessionStorageName) as string;
+    if (kubeconfig === null) {
+      return null;
+    }
+    const conf: Kubeconfig = loadYaml(kubeconfig);
+    const contexts = conf.contexts.filter(ctx => {
+      return ctx.name === contextName;
+    });
+    if (!contexts.length) {
+      return null;
+    }
+    return contexts[0];
+  }
+
+  getCluster(name: string): Cluster {
+    const kubeconfig = this.sessionStorage_.retrieve(this.CONFIG.kubeconfigSessionStorageName) as string;
+    if (kubeconfig === null) {
+      return null;
+    }
+    const conf: Kubeconfig = loadYaml(kubeconfig);
+    const clusters = conf.clusters.filter(cluster => {
+      return cluster.name === name;
+    });
+    if (!clusters.length) {
+      return null;
+    }
+    return clusters[0];
+  }
+
+  getCurrentServer(): string {
+    const contextName = this.getCurrentContext();
+    return this.getServer(contextName);
+  }
+
+  getServer(contextName: string): string {
+    // Get Context
```

Comment:
These comments are not required, it's easy to see what is this code is doing by looking at it. Remove.